include/boost_plugin_loader/plugin_loader.hpp

```diff
@@ -56,6 +56,30 @@ static std::shared_ptr<ClassBase> createSharedInstance(const std::string& symbol
   return std::shared_ptr<ClassBase>(plugin.get(), [plugin](ClassBase*) mutable { plugin.reset(); });
 }
 
+/**
+ * @brief This will remove libraries with full path in the provided library_names and return them.
+ * @param library_names The set to search and remove libraries with full paths
+ * @return A set of the libraries provided as full path
+ */
+inline std::set<std::string> extractLibrariesWithFullPath(std::set<std::string>& library_names)
+{
+  std::set<std::string> libraries_with_fullpath;
+  for (auto it = library_names.begin(); it != library_names.end();)
+  {
+    if (boost::filesystem::exists(*it))
```

Comment:
This should also check to make sure the file is an absolute path using `boost::filesystem::is_absolute() `